### ISSUE TITLE
Panic with pointer-to-interface field and collection value

### ISSUE THREAD

**Issue by sbinq** [mitchellh/copystructure#19]
Issue similar to #16 (and probably even more artificial than what was described there - my example is _not_ based on production code):

```
package main

import (
	"testing"

	"github.com/mitchellh/copystructure"
	"github.com/stretchr/testify/require"
)

func TestCopyWithPointerToInterfaceFieldWithSliceValue(t *testing.T) {
	type Struct struct {
		IfacePtr *interface{}
	}

	ifacePtr := func(v interface{}) *interface{} {
		return &v
	}

	copy, err := copystructure.Copy(Struct{IfacePtr: ifacePtr([]string{"111"})})
	require.Nil(t, err)
	require.Equal(t, Struct{IfacePtr: ifacePtr([]string{"111"})}, copy)
}

func TestCopyWithPointerToInterfaceFieldWithMapValue(t *testing.T) {
	type Struct struct {
		IfacePtr *interface{}
	}

	ifacePtr := func(v interface{}) *interface{} {
		return &v
	}

	copy, err := copystructure.Copy(Struct{IfacePtr: ifacePtr(map[string]string{"a": "111"})})
	require.Nil(t, err)
	require.Equal(t, Struct{IfacePtr: ifacePtr(map[string]string{"a": "111"})}, copy)
}
```

Output:
```
➜  src go test -run TestCopyWithPointerToInterfaceFieldWithSliceValue
--- FAIL: TestCopyWithPointerToInterfaceFieldWithSliceValue (0.00s)
panic: reflect.Set: value of type []string is not assignable to type *interface {} [recovered]
	panic: reflect.Set: value of type []string is not assignable to type *interface {}

goroutine 5 [running]:
panic(0x63a9a0, 0xc420013650)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
testing.tRunner.func1(0xc420090180)
	/usr/local/go/src/testing/testing.go:579 +0x25d
panic(0x63a9a0, 0xc420013650)
	/usr/local/go/src/runtime/panic.go:458 +0x243
reflect.Value.assignTo(0x637440, 0xc4200d6da0, 0x97, 0x697523, 0xb, 0x62e280, 0x0, 0x1, 0x1, 0x0)
	/usr/local/go/src/reflect/value.go:2163 +0x35c
reflect.Value.Set(0x62e280, 0xc420032058, 0x196, 0x637440, 0xc4200d6da0, 0x97)
	/usr/local/go/src/reflect/value.go:1333 +0xa4
github.com/mitchellh/copystructure.(*walker).Exit(0xc420098090, 0x7, 0x0, 0x0)
	/home/sbinq/work/gopath/vendor/src/github.com/mitchellh/copystructure/copystructure.go:202 +0x86a
github.com/mitchellh/reflectwalk.walkStruct(0x6568c0, 0xc4200135b0, 0x19, 0x687900, 0xc420098090, 0x0, 0x10)
	/home/sbinq/work/gopath/vendor/src/github.com/mitchellh/reflectwalk/reflectwalk.go:330 +0x40c
github.com/mitchellh/reflectwalk.walk(0x6568c0, 0xc4200135b0, 0x19, 0x687900, 0xc420098090, 0x0, 0x0)
	/home/sbinq/work/gopath/vendor/src/github.com/mitchellh/reflectwalk/reflectwalk.go:180 +0x4b2
github.com/mitchellh/reflectwalk.Walk(0x6568c0, 0xc4200135b0, 0x687900, 0xc420098090, 0x7be4a0, 0xc42002feb8)
	/home/sbinq/work/gopath/vendor/src/github.com/mitchellh/reflectwalk/reflectwalk.go:84 +0x155
github.com/mitchellh/copystructure.Config.Copy(0x647d00, 0x0, 0x6568c0, 0xc4200135b0, 0x47679d, 0x647d40, 0xc4200135b0, 0xc4200135b0)
	/home/sbinq/work/gopath/vendor/src/github.com/mitchellh/copystructure/copystructure.go:69 +0x81
github.com/mitchellh/copystructure.Copy(0x6568c0, 0xc4200135b0, 0xc4200135b0, 0x637440, 0xc4200d6d00, 0x7be4a0)
	/home/sbinq/work/gopath/vendor/src/github.com/mitchellh/copystructure/copystructure.go:13 +0x4c
_/home/sbinq/work/gopath/src.TestCopyWithPointerToInterfaceFieldWithSliceValue(0xc420090180)
	/home/sbinq/work/gopath/src/sample_test.go:19 +0xc8
testing.tRunner(0xc420090180, 0x6b6f68)
	/usr/local/go/src/testing/testing.go:610 +0x81
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:646 +0x2ec
exit status 2
FAIL	_/home/sbinq/work/gopath/src	0.005s
➜  src go test -run TestCopyWithPointerToInterfaceFieldWithMapValue  
--- FAIL: TestCopyWithPointerToInterfaceFieldWithMapValue (0.00s)
panic: reflect.Set: value of type map[string]string is not assignable to type *interface {} [recovered]
	panic: reflect.Set: value of type map[string]string is not assignable to type *interface {}

goroutine 5 [running]:
panic(0x63a9a0, 0xc420013670)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
testing.tRunner.func1(0xc420090180)
	/usr/local/go/src/testing/testing.go:579 +0x25d
panic(0x63a9a0, 0xc420013670)
	/usr/local/go/src/runtime/panic.go:458 +0x243
reflect.Value.assignTo(0x64d1e0, 0xc4200154d0, 0x15, 0x697523, 0xb, 0x62e280, 0x0, 0x1, 0x1, 0x0)
	/usr/local/go/src/reflect/value.go:2163 +0x35c
reflect.Value.Set(0x62e280, 0xc420032058, 0x196, 0x64d1e0, 0xc4200154d0, 0x15)
	/usr/local/go/src/reflect/value.go:1333 +0xa4
github.com/mitchellh/copystructure.(*walker).Exit(0xc420098090, 0x7, 0x0, 0x0)
	/home/sbinq/work/gopath/vendor/src/github.com/mitchellh/copystructure/copystructure.go:202 +0x86a
github.com/mitchellh/reflectwalk.walkStruct(0x656940, 0xc4200135b0, 0x19, 0x687900, 0xc420098090, 0x0, 0x10)
	/home/sbinq/work/gopath/vendor/src/github.com/mitchellh/reflectwalk/reflectwalk.go:330 +0x40c
github.com/mitchellh/reflectwalk.walk(0x656940, 0xc4200135b0, 0x19, 0x687900, 0xc420098090, 0x0, 0x0)
	/home/sbinq/work/gopath/vendor/src/github.com/mitchellh/reflectwalk/reflectwalk.go:180 +0x4b2
github.com/mitchellh/reflectwalk.Walk(0x656940, 0xc4200135b0, 0x687900, 0xc420098090, 0xc420015440, 0xc42002fef8)
	/home/sbinq/work/gopath/vendor/src/github.com/mitchellh/reflectwalk/reflectwalk.go:84 +0x155
github.com/mitchellh/copystructure.Config.Copy(0x647d00, 0x0, 0x656940, 0xc4200135b0, 0x47682d, 0x647d40, 0xc4200135b0, 0xc4200135b0)
	/home/sbinq/work/gopath/vendor/src/github.com/mitchellh/copystructure/copystructure.go:69 +0x81
github.com/mitchellh/copystructure.Copy(0x656940, 0xc4200135b0, 0xc4200135b0, 0x6b8570, 0xc420015440, 0x30)
	/home/sbinq/work/gopath/vendor/src/github.com/mitchellh/copystructure/copystructure.go:13 +0x4c
_/home/sbinq/work/gopath/src.TestCopyWithPointerToInterfaceFieldWithMapValue(0xc420090180)
	/home/sbinq/work/gopath/src/sample_test.go:33 +0xec
testing.tRunner(0xc420090180, 0x6b6f58)
	/usr/local/go/src/testing/testing.go:610 +0x81
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:646 +0x2ec
exit status 2
FAIL	_/home/sbinq/work/gopath/src	0.005s
➜  src 
```

**Comment by sbinq** [comment]
Seems like #24 fixes this issue too (please ignore first PR mentioned here - occasional mistake).

**Comment by mitchellh** [comment]
Fixed by #24
